src/crystal/system/unix/syscall/linux.cr

```diff
@@ -0,0 +1,291 @@
+    property sq_thread_cpu = 0u32
+    property sq_thread_idle = 0u32
+    property features = IoUringFeatures::None
+    property wq_fd = 0u32
+    property resv = StaticArray(UInt32, 3).new(0)
+    property sq_off = IoSqringOffsets.new
+    property cq_off = IoCqringOffsets.new
+  end
+
+  @[Extern]
+  struct IoUringCqe
+    property user_data = 0u64 # sqe->data submission passed back
+    property res = 0i32       # result code for this event
+    property flags = 0u32
+  end
+
+  @[Flags]
+  enum RwFlag : UInt32
+    HIPRI
+    DSYNC
+    SYNC
+    NOWAIT
+    APPEND
+  end
+
+  @[Flags]
+  enum PollEvent : UInt32
+    IN     = 0x0001
+    PRI    = 0x0002
+    OUT    = 0x0004
+    ERR    = 0x0008
+    HUP    = 0x0010
+    NVAL   = 0x0020
+    RDNORM = 0x0040
+    RDBAND = 0x0080
+    WRNORM = 0x0100
+    WRBAND = 0x0200
+    MSG    = 0x0400
+    REMOVE = 0x1000
+    RDHUP  = 0x2000
+  end
+
+  @[Extern(union: true)]
+  struct IoUringSqeInnerFlags
+    property rw_flags = RwFlag::None
+    property poll_events = PollEvent::None
+    # property fsync_flags = 0u32
+    # property sync_range_flags = 0u32
+    # property msg_flags = 0u32
+    # property timeout_flags = 0u32
+    # property accept_flags = 0u32
+    # property cancel_flags = 0u32
+    # property open_flags = 0u32
+    # property statx_flags = 0u32
+    # property fadvise_advice = 0u32
+    # property splice_flags = 0u32
+    # property rename_flags = 0u32
+    # property unlink_flags = 0u32
+  end
+
+  @[Extern]
+  struct IoUringSqe
+    property opcode = IoUringOp::NOP       # type of operation for this sqe
+    property flags = IoUringSqeFlags::None # flags
+    property ioprio = 0u16                 # ioprio for the request
+    property fd = 0i32                     # file descriptor to do IO on
+    property off = 0u64                    # offset into file
+    property addr = 0u64                   # pointer to buffer or iovecs
+    property len = 0u32                    # buffer size or number of iovecs
+    property inner_flags = IoUringSqeInnerFlags.new
+    property user_data = 0u64          # data to be passed back at completion time
+    property buf_index_or_group = 0u16 # index into fixed buffers, if used, or for grouped buffer selection
+    property personality = 0u16
+    property splice_fd_in = 0i32
+    property pad1 = 0u64
+    property pad2 = 0u64
+  end
+
+  @[Flags]
+  enum IoUringProbeOpFlags : UInt16
+    SUPPORTED
+  end
+
+  @[Extern]
+  struct IoUringProbeOp
+    property op = 0u8
+    property resv = 0u8
+    property flags = IoUringProbeOpFlags::None
+    property resv2 = 0u32
+  end
+
+  @[Extern]
+  struct IoUringProbe(N)
+    property last_op = 0u8 # last opcode supported
+    property ops_len = 0u8 # length of ops[] array below
+    property resv = 0u16
+    property resv2 = StaticArray(UInt32, 3).new(0)
+    property ops = StaticArray(IoUringProbeOp, N).new(IoUringProbeOp.new)
+  end
+
+  @[Extern]
+  record IOVec,
```

Comment:
This one (and msg_header, later when the method that needs that is implemented), should probably be in `LibC`.